### ISSUE TITLE
bilge generates false positive errors in rust_analyzer

### ISSUE THREAD

**Issue by denniskempin** [hecatia-elegua/bilge#92]
In a dead simple example:

```
use bilge::prelude::*;

#[bitsize(8)]
#[derive(FromBits)]
struct Test {
    a: u4,
    b: u4,
}
```

The latest version (0.3.2020) of rust_analyzer reports:

```
❯ rust-analyzer diagnostics .
processing crate: foo, module: /Users/denniskempin/workspace/test/foo/src/main.rs
Diagnostic { code: Ra("macro-error", Error), message: "add #[bitsize] attribute above your derive attribute", range: FileRange { file_id: FileId(11), range: 24..37 }, severity: Error, unused: false, experimental: false, fixes: None, main_node: None }
```

i.e.

`add #[bitsize] attribute above your derive attribute` from this [piece of code](https://github.com/hecatia-elegua/bilge/blob/a338e8287a28375682c8224e675d2bf72bb8e78b/bilge-impl/src/shared.rs#L52)

Oddly enough, the same error does not appear with `cargo check`.

**Comment by hecatia-elegua** [comment]
They might have done something to the ordering of macro execution.
I thought there is a silent agreement about that, but if not... well... it still works on the normal compiler.
Is it always happening? If yes, I would have to ask the ra devs somewhere

**Comment by showier-drastic** [comment]
Just met the same problem. Reproducible with very simple projects.

**Comment by showier-drastic** [comment]
FYI RA devs [replied](https://github.com/rust-lang/rust-analyzer/issues/17873#issuecomment-2367009286).

**Comment by alexkl-rugged-controls** [comment]
It looks like `rust-analyzer` re-evaluates `derive` macros once for every macro defined below it: https://github.com/rust-lang/rust-analyzer/issues/17873#issuecomment-2495081092

**Comment by hecatia-elegua** [comment]
Thank you both.

**Comment by TimeTravelPenguin** [comment]
Is there currently a workaround for this? If not, do you have any idea as to when the issue will be resolved? Thanks for your excellent work :)

**Comment by alextechcc** [comment]
Yeah (sorry, same person here that posted explanation), I have no idea why rust-analyzer re-evaluates every single derive for every single non-derive. It seems to me that it could be a critical performance issue, but I'm not sure. For working around it - you'd need some way to flag that something has already been processed and not process it again - seems a lot easier if you're in rust-analyzer land.

**Comment by bokken12** [comment]
As a workaround, I find I can largely avoid this issue by gating like
```
#[cfg_attr(not(rust_analyzer), bitsize(16))]
```

**Comment by widberg** [comment]
The RA issue was closed as complete, and after updating, I'm not seeing these errors in my editor where I used to. I'm going to close this as completed, but feel free to open a new issue with RA if the problem persists.
